sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreCheckpointFormatV2Suite.scala

```diff
@@ -305,6 +597,252 @@ class RocksDBStateStoreCheckpointFormatV2Suite extends StreamTest
+      // Use stateUniqueIds(0) because there is only one state operator
+      val res2 = metadata.stateUniqueIds(0).map { uniqueIds =>
+        uniqueIds(0)
+      }
+      idx -> res2
+    }.toMap
+
+    versionToUniqueIdFromCommitLog.foreach {
+      case (version, uniqueIds) =>
+        if (!versionToUniqueIdFromStateStore(version).sorted.sameElements(uniqueIds.sorted)) {
+          ret = false
+          return ret
+        }
+    }
+    ret
+  }
+
+  // scalastyle:off line.size.limit
+  // When there is task retries or multiple jobs launched, there will be different
+  // sets of state store checkpointInfo because of lineage.
+  // e.g.
+  // [Picked] StateStoreCheckpointInfo[partitionId=1, batchVersion=2, stateStoreCkptId=Some(a9d5afec-0e8d-4473-b948-6c55513aa509), baseStateStoreCkptId=Some(061f7c53-b300-477a-a599-5387d55e315a)]
+  // [Picked] StateStoreCheckpointInfo[partitionId=0, batchVersion=2, stateStoreCkptId=Some(879cc517-6b85-4dae-abba-794bf2dbab82), baseStateStoreCkptId=Some(513726e7-2448-41a6-a874-92053c5cf86b)]
+  // StateStoreCheckpointInfo[partitionId=1, batchVersion=2, stateStoreCkptId=Some(7f4ad39f-d019-4ca2-8cf4-300379821cd6), baseStateStoreCkptId=Some(061f7c53-b300-477a-a599-5387d55e315a)]
+  // StateStoreCheckpointInfo[partitionId=0, batchVersion=2, stateStoreCkptId=Some(9dc215fe-54f9-4dc1-a59b-a8734f359e46), baseStateStoreCkptId=Some(513726e7-2448-41a6-a874-92053c5cf86b)]
+  // This function finds the correct one based on commit log
+  // scalastyle:on line.size.limit
+   private def pickCheckpointInfoFromCommitLog(
+        checkpointDir: File,
+        checkpointInfoList: Iterable[StateStoreCheckpointInfo]
+        ): Option[Iterable[StateStoreCheckpointInfo]] = {
+      val allPossibleCheckpointInfoList = checkpointInfoList
+        .groupBy(x => (x.partitionId, x.batchVersion)).values.transpose
+
+     var ret: Option[Iterable[StateStoreCheckpointInfo]] = None
+     for (elem <- allPossibleCheckpointInfoList) {
+       if (verifyCheckpointInfoFromCommitLog(checkpointDir, elem)) {
+         ret = Some(elem)
+         return ret
+       }
+     }
+     ret
+   }
+
+  // This test verifies when there are two job launched inside the
+  // foreachBatch function (write to multiple sinks).
+  // When this happens, the unique ids actually stored in the commit log should be the the same
+  // as those recorded by CkptIdCollectingStateStoreWrapper
+  testWithCheckpointInfoTracked(s"checkpointFormatVersion2 validate ID - two jobs launched") {
+    withTempDir { checkpointDir =>
+      withTable("wei_test_t1") {
+        withTable("wei_test_t2") {
+          val inputData = MemoryStream[Int]
+          val aggregated =
+            inputData
+              .toDF()
+              .groupBy($"value")
+              .agg(count("*"))
+
+          val writer = (ds: DataFrame, batchId: Long) => {
+            ds.write.mode("append").save("wei_test_t1")
+            ds.write.mode("append").save("wei_test_t2")
```

Comment:
please write to a temp directory and delete them. These are currently not creating tables. Or use `saveAsTable`